controllers/front/notify.php

```diff
@@ -130,6 +140,8 @@ public function postProcess()
             $updateOrderStatusAction = $this->module->getService(UpdateOrderStatusAction::class);
             $updateOrderStatusAction->run((int) $orderId, (int) Configuration::get('SAFERPAY_PAYMENT_AUTHORIZED'));
 
+            $this->releaseLock();
```

Comment:
release lock on `finally` only, additionally there should be an option for autorelease? so this isn't really needed.